packages/template-retail-react-app/app/commerce-api/hooks/useCustomerProductLists.js

```diff
@@ -263,6 +263,28 @@ export default function useCustomerProductLists() {
                         allImages: true
                     }
                 })
+
+                // `getProducts` endpoint does not include `setProducts` data
+                // so for each product set, we'll fetch it and append it to existing data
+                const productSetFetches = []
+                productDetails.data
+                    .filter((product) => product.type.set)
+                    .forEach((product) => {
+                        const promise = api.shopperProducts
+                            .getProduct({
+                                parameters: {
+                                    id: product.id
+                                }
+                            })
+                            .then((data) => {
+                                product.setProducts = data.setProducts
+                            })
+
+                        productSetFetches.push(promise)
+                    })
+
+                await Promise.allSettled(productSetFetches)
```

Comment:
This ignores fetches that fail. What happens when some products have `setProducts` but others don't?

Reply:
Good question.. it should be `Promise.all`. 

I was thinking of this extending the data with fetched `setProducts` as something "optional". But actually, that was not accurate.